### ISSUE TITLE
Decide if "Re-analyse beats" should be ON by default for 1.11

### ISSUE THREAD

**Issue by Holzhaus** [Holzhaus/mixxx-gh-issue-migration#439]
Now we got the praised QM tempo beat detection + improved syncing features in 1.11 there is a catch: All the syncing stuff rises and falls with correct beat detection data for the tracks playing. This correlation is not apparent to the user.

Users upgrading from previous versions won't profit until they discover the "Re-analyse beats" option. Until then we`ll still hear trough the forums and feedback form how much beat detection sucks, that loops and and sync drift .

How about detecting if a user is upgrading and presenting a chooser dialog with additional info message why they should re-analyze the files?

Setting the option to ON dy default could backfire for users who have neatly set beatgrids by hand.

**Comment by Holzhaus** [comment]
(by rryan)
I agree... we really need an upgrade dialog otherwise people will miss it. I don't want to automatically opt them in.

**Comment by Holzhaus** [comment]
(by ywwg)
Would clicking OK on the dialog actually trigger the analysis or just toggle the checkbox so files are reanalyzed on load?

**Comment by Holzhaus** [comment]
(by rryan)
It would be a one-time question on the first run of 1.11.0 that
enables/disables the checkbox.

**Comment by Holzhaus** [comment]
(by ywwg)
I'll bang this out tonight.

**Comment by Holzhaus** [comment]
(by rryan)
Thanks! For reference src/upgrade.cpp is where all the config-upgrade code
lives now.

**Comment by Holzhaus** [comment]
(by ywwg)
Here's the patch.  I spent a lot of time on the wording, trying to make it clear that upgrading beatgrids was a good idea, and the default choice, but the user has the option of opting out.

**Comment by Holzhaus** [comment]
(by jus)
Thanks for the patch.

* Kinda miss a hint, that this message is only displayed once because a user is upgrading from a previous version of Mixxx? What about " Upgrading Mixxx from v1.9.x/1.10.x." in the window title?
* The catchphrase should be revised imo, better describe the benefit. Users (mostly) are not programmers, they don't care if Mixxx has a new beat detection library. What about "Improved beat and tempo detection in Mixxx"
* What about a hint, that re-analyzing beatgrids will overwrite old beat detection data but won't affect Cue-points, Hotcue, Playlists , Crates and the audio files itself. If this is clear to the user, they more likely opt-in for the benefits (i would).
* Settle for " beat grids" or "beatgrids" in the text
* Is the "?" icon in the message by purpose, can we display the Mixxx icon instead (see screenshot) ? The window did not get focus automatically, it would be more clear it belongs to Mixxx. Alternatively, we can use the icon in the window title like with the main application window.

Should we automatically make a backup copy of mixxxdb.sqlite before upgrading just in case?

**Comment by Holzhaus** [comment]
(by jus)

**Comment by Holzhaus** [comment]
(by rryan)
I played with the wording a little bit here.

RE: Jus 

* I changed the Window title as you suggested but don't see any change on OS X.
* Agree RE: catchphrase. I made it "Mixxx has a new and improved beat detector"
* Added a hint that hotcues/cues/playlists/crates aren't affected. I don't know if it's worth mentining the file itself won't be changed.
* Changed to "beatgrid"
* Changed to use the Mixxx icon.

I'm against making a backup copy of mixxxdb.sqlite. We didn't get any feedback from someone that was unhappy about us losing their metadata in the beta period which leads me to believe either we warned people well enough via the forum post or they just don't care :).

**Comment by Holzhaus** [comment]
(by jus)
Hmm, i cant't see a window title either in OSX.
Anyway, lets go with the latest patch and an updated translation template. I`ll and drop a note to launchpad-translators so at least these important strings are translated for 1.11 into as many languages ​​as possible.

One little thing: On OSX, the whole text in the message box is displayed in bold, so the part inside the <b></b> tag is not emphasized. Maybe just change it to <u></u>.

**Comment by Holzhaus** [comment]
(by rryan)
I just removed the bolding.

Also, I just discovered the --author flag to bzr commit which looks like it gives Owen credits in the log:

revno: 3724
fixes bug: https://launchpad.net/bugs/1093431
author: Owen Williams <email address hidden>
committer: RJ Ryan <email address hidden>
branch nick: 1.11
timestamp: Sat 2013-02-09 13:21:35 -0500
message:
  Add upgrade dialog box to ask whether to turn on re-analyze beats. Fixes Bug #1093431

We should use this flag in the future when committing patches from people who don't have commit access so they can get credits in the log.
